components/ShadowLayer/src/MDCShadowLayer.m

```diff
@@ -359,33 +367,15 @@ - (void)commonLayoutSublayers {
 
 @end
 
-@implementation MDCAnimatedShapeLayer
-
-- (id<CAAction>)actionForKey:(NSString *)key {
-  if ([key isEqualToString:@"path"] || [key isEqualToString:@"shadowPath"]) {
-    // We have to create a pending animation because if we are inside a UIKit animation block we
-    // won't know any properties of the animation block until it is commited.
-    MDCPendingAnimation *pendingAnim = [[MDCPendingAnimation alloc] init];
-    pendingAnim.fromValue = [self.presentationLayer valueForKey:key];
-    pendingAnim.toValue = nil;
-    pendingAnim.keyPath = key;
-
-    return pendingAnim;
-  }
-  return [super actionForKey:key];
-}
-
-@end
-
 @implementation MDCPendingAnimation
 
 - (void)runActionForKey:(NSString *)event object:(id)anObject arguments:(NSDictionary *)dict {
-  if ([anObject isKindOfClass:[MDCAnimatedShapeLayer class]]) {
-    MDCAnimatedShapeLayer *layer = (MDCAnimatedShapeLayer *)anObject;
+  if ([anObject isKindOfClass:[CAShapeLayer class]]) {
+    CAShapeLayer *layer = (CAShapeLayer *)anObject;
 
     // In order to synchronize our animation with UIKit animations we have to fetch the resizing
     // animation created by UIKit and copy the configuration to our custom animation.
-    CALayer *animatedLayer = layer.animationParentLayer;
+    CALayer *animatedLayer = (CALayer *)layer.delegate;
```

Comment:
This is depending on an implementation detail here and won’t be caught if someone changes the delegate in the code above. Can you get access to the parent layer by storing it in the pending animation?

Reply:
Good point. Done.